server/transport.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Timothy Chen
+
+package server
+
+import (
+	"net"
+	"net/rpc"
+	"sync"
+
+	"github.com/cockroachdb/cockroach/gossip"
+	"github.com/cockroachdb/cockroach/multiraft"
+	"github.com/cockroachdb/cockroach/proto"
+	crpc "github.com/cockroachdb/cockroach/rpc"
+	"github.com/cockroachdb/cockroach/rpc/codec"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+const (
+	raftServiceName = "MultiRaft"
+	raftMessageName = raftServiceName + ".RaftMessage"
+)
+
+// RPCTransport handles the rpc messages for multiraft.
+type RPCTransport struct {
```

Comment:
Probably not. It's possible that we'd want to use this in place of LocalRPCTransport in some integration-style tests, but given the uncertainty of gossip propagation I doubt we'd want to use it outside of this package. I'll un-export it unless and until we have another use for it.